R/phylo_check.R

```diff
@@ -0,0 +1,133 @@
+#' Check phylogeny tip names
+#'
+#' A function to check the list of tip names in a phylogeny against a vector of
+#' taxon names, and if desired, to trim the phylogeny to only include taxon
+#' names within the vector.
+#'
+#' @param tree \code{phylo}. A phylo object. Phylogenies can be read into R from
+#' .txt or .tree files containing the Newick formatted tree using
+#' \code{ape::read.tree}.
+#' @param list \code{character}. A vector of taxon names. Binomials can be
+#' separated with either a space or an underscore. The list should not contain
+#' any punctuation.
+#' @param out \code{character}. Determine whether to return either the counts
+#' of taxa included and not included in the tree ("counts", the default), a
+#' \code{dataframe} describing which taxa are included or not included in the
+#' tree ("table"), or the phylogeny trimmed to only include taxa in the provided
+#' list ("tree").
+#' @return If out = "counts", a summary table containing the number of taxa in
+#' the list but not the tree, in the tree but not the list, and in both. If out
+#'  = "table", a \code{dataframe} describing whether taxon names are present in
+#' the list and/or the tree. If out = "tree", the input phylogeny trimmed to
+#' only include the tips present in the list; you can save your new tree using
+#' \code{ape::write.tree}.
+#' @importFrom ape drop.tip
+#' @section Developer(s):
+#' Bethany Allen
+#' @section Reviewer(s):
+#'
+#' @examples
+#' #Read in example tree of ceratopsians from paleotree
+#' library(paleotree)
+#' data(RaiaCopesRule)
+#' plot(ceratopsianTreeRaia)
+#'
+#' #Specify list of names
+#' dinosaurs <- c("Nasutoceratops_titusi", "Diabloceratops_eatoni",
+#' "Zuniceratops_christopheri", "Psittacosaurus_major",
+#' "Psittacosaurus_sinensis", "Avaceratops_lammersi",
+#' "Xenoceratops_foremostensis", "Leptoceratops_gracilis",
+#' "Triceratops_horridus", "Triceratops_prorsus")
+#'
+#' #Counts of taxa in list, tree or both
+#' phylo_check(tree = ceratopsianTreeRaia, list = dinosaurs)
+#'
+#' #Table of taxon names in list, tree or both
+#' phylo_check(tree = ceratopsianTreeRaia, list = dinosaurs, out = "table")
+#'
+#' #Trim tree to tips in the list
+#' my_ceratopsians <- phylo_check(tree = ceratopsianTreeRaia, list = dinosaurs,
+#' out = "tree")
+#' plot(my_ceratopsians)
+#' @export
+
+phylo_check <- function(tree = NULL, list = NULL, out = "counts") {
```

Comment:
I wonder if the default for `out` should be "table"? That seems the most useful to me.

Reply:
That's reasonable. I have changed this and rearranged the documentation accordingly.